source/opt/ir_context.h

```diff
@@ -170,6 +173,15 @@ class IRContext {
     return (entry != instr_to_block_.end()) ? entry->second : nullptr;
   }
 
+  // Returns a pointer the decoration manager.  If the decoration manger is
+  // invalid, it is rebuilt first.
+  opt::analysis::DecorationManager* get_decoration_mgr() {
```

Comment:
Not to be resolved in this MR.  But why are we returning pointers when we could be returning references?   The result is never going to be null.

Reply:
I'll keep it consistent with the other analyses for now.  Then we can change them after.